tests/test_priority_samples_process.py

```diff
@@ -0,0 +1,379 @@
+            },
+            # Process some, different plates, one plate running, one plate running nothing to process
+            {
+                "processed_status": [False, False, False, True],
+                "plate_barcodes": ["123", "123", "123", "456"],
+                "plates_status": {"123": DART_STATE_PENDING, "456": "RUNNING"},
+                "expected_mlwh_samples": [True, True, True, False],
+                "expected_dart_samples": [True, True, True, False],
+                "expected_dart_plates": ["123"],
+            },
+            # Process all, different plates, all plates running, so no process
+            {
+                "processed_status": [False, False, False, False],
+                "plate_barcodes": ["123", "123", "123", "456"],
+                "plates_status": {"123": "RUNNING", "456": "RUNNING"},
+                "expected_mlwh_samples": [True, True, True, True],
+                "expected_dart_samples": [False, False, False, False],
+                "expected_dart_plates": ["123", "456"],
+            },
+        ]
+    )
+    def with_different_scenarios(self, request, mongo_database, testing_samples, testing_priority_samples):
+        _, mongo_database = mongo_database
+
+        # Set processed samples
+        self.processed_status = request.param["processed_status"]
+        self.set_property_values_for_collection(
+            mongo_database,
+            COLLECTION_PRIORITY_SAMPLES,
+            testing_priority_samples,
+            FIELD_PROCESSED,
+            self.processed_status,
+        )
+
+        # Set plate barcodes for each sample
+        self.plate_barcodes = request.param["plate_barcodes"]
+        self.set_property_values_for_collection(
+            mongo_database, COLLECTION_SAMPLES, testing_samples, FIELD_PLATE_BARCODE, self.plate_barcodes
+        )
+
+        # Set plate status for each plate and mock calls
+        self.plate_status = request.param["plates_status"]
+
+        def find_plate_status(cursor, plate_barcode, lab_type):
+            return self.plate_status[plate_barcode]
+
+        self.mock_add_dart_plate.side_effect = find_plate_status
+
+        def extract_mongo_record(info: RecordScenarioList) -> SampleDocValue:
+            return info[0]
+
+        # Set expected mlwh samples
+        self.expected_mlwh_samples = list(
+            map(
+                extract_mongo_record,
+                filter(lambda info: info[1], zip(testing_priority_samples, request.param["expected_mlwh_samples"])),
+            )
+        )
+
+        # Set expected dart samples
+        self.expected_dart_samples = list(
+            map(
+                extract_mongo_record,
+                filter(lambda info: info[1], zip(testing_samples, request.param["expected_dart_samples"])),
+            )
+        )
+
+        # Set expected dart plates
+        self.expected_dart_plates = request.param["expected_dart_plates"]
+
+    # Currently not being ran
```

Comment:
What is not being run?

Reply:
Its a test to test the generation of the test data fixture  - whoa! We couldn't get it working but will have another go